BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/BeaconDistanceCalculator.java

```diff
@@ -30,12 +30,16 @@ public static float calculateDistanceTo(Beacon beacon) {
      * rather than the altitude above sea level.
      */
     public static float calculateDistanceWithoutAltitudeDeltaToFloor(Beacon beacon, float rssi) {
-        double altitude = beacon.getLocation().getAltitude();
         float distance = calculateDistanceTo(beacon, rssi);
-        // distance should be double of the altitude to make pythagoras meaningful
-        if (altitude > 0 && distance > (altitude * 2)) {
-            double delta = Math.pow(distance, 2) - Math.pow(altitude, 2);
-            return (float) Math.sqrt(delta);
+        if (beacon.hasLocation() && beacon.getLocation().hasElevation()) {
```

Comment:
The device elevation is missing in this calculation (you currently assume it's 0). It's required to actually calculate the elevation delta, which should be used instead of the beacon elevation.

Reply:
resolved